stories/3-Button.stories.tsx

```diff
@@ -6,106 +6,122 @@ import FileUploadButton from "../src/button/file-upload/FileUploadButton";
-      customSpinner={<SpinnerStorySample />}>
-      {"Click Me - shouldDisplaySpinner - customSpinner"}
-    </Button>
-
-    <br />
-    <hr />
-    <br />
-
-    <FileUploadButton
-      onFileSelect={(files) =>
-        alert(
-          Array.from(files)
-            .map((file) => file.name)
-            .join(", ")
-        )
-      }
-      name={"photos"}
-      htmlFor={"photos"}>
-      {"Upload your photos"}
-    </FileUploadButton>
-
-    <br />
-
-    <FileUploadButton
-      onFileSelect={(files) =>
-        alert(
-          Array.from(files)
-            .map((file) => file.name)
-            .join(", ")
-        )
-      }
-      name={"second-photos"}
-      htmlFor={"second-photos"}
-      isDisabled={true}>
-      {"Upload your photos - isDisabled"}
-    </FileUploadButton>
-
-    <br />
-
-    <FileUploadButton
-      onFileSelect={(files) =>
-        alert(
-          Array.from(files)
-            .map((file) => file.name)
-            .join(", ")
-        )
-      }
-      name={"second-photos"}
-      htmlFor={"second-photos"}
-      isPending={true}>
-      {"Upload your photos - isPending"}
-    </FileUploadButton>
-
-    <br />
-
-    <FileUploadButton
-      onFileSelect={(files) =>
-        alert(
-          Array.from(files)
-            .map((file) => file.name)
-            .join(", ")
-        )
-      }
-      name={"second-photos"}
-      htmlFor={"second-photos"}
-      isPending={true}
-      customSpinner={<SpinnerStorySample />}>
-      {"Upload your photos - isPending - customSpinner"}
-    </FileUploadButton>
-  </StoryFragment>
-));
+import {useRef} from "@storybook/addons";
+
+storiesOf("Button", module).add("Button States", () => {
+  const buttonRef = useRef<HTMLButtonElement | null>(null);
+
+  return (
+    <StoryFragment>
+      <Button type={"button"} onClick={(e) => alert("Thank You!")}>
+        {"Click Me"}
+      </Button>
+
+      <br />
+
+      <Button type={"button"} onClick={(e) => alert("Thank You!")} isDisabled={true}>
+        {"Click Me - isDisabled"}
+      </Button>
+
+      <br />
+
+      <StateProvider initialState={false}>
+        {(state, setState) => (
+          <Button
+            type={"button"}
+            onClick={() => setState(true)}
+            shouldDisplaySpinner={state}>
+            {"Click Me - shouldDisplaySpinner"}
+          </Button>
+        )}
+      </StateProvider>
+
+      <br />
+
+      <Button
+        type={"button"}
+        onClick={(e) => alert("Thank You!")}
+        shouldDisplaySpinner={true}
+        customSpinner={<SpinnerStorySample />}>
+        {"Click Me - shouldDisplaySpinner - customSpinner"}
+      </Button>
+
+      <br />
+
+      <Button
+        ref={buttonRef}
+        type={"button"}
+        onClick={() =>
+          alert(`Button width: ${buttonRef.current.getBoundingClientRect().width}`)
+        }>
+        {"Get Button Width"}
+      </Button>
+
+      <br />
+      <hr />
+      <br />
+
+      <FileUploadButton
```

Comment:
Can we create a story for FileUploadButton examples ?

Reply:
For use `ref` in the `FileUploadButton` as well, we should create with `forwardRef`. I don't think so we need this. But you think there is any use-case, we can add.

Reply:
I'll add the functionality it needs, now. 👍🏻

Reply:
I think we should separate regular button stories from file upload button examples